packages/driver/src/cy/options.ts

```diff
@@ -0,0 +1,47 @@
+import _ from 'lodash'
+
+export const getIndexOfUserOptions = (name: string, args: any[] | undefined): number => {
+  if (!args || args.length === 0) return -1
+
+  const lastIndex = args.length - 1
+  const lastArg = args[lastIndex]
+
+  // Handle exceptions
+  if (name === 'invoke') {
+    return args.length >= 2 && !_.isFunction(args[0]) ? 0 : -1
+  }
+
+  if (name === 'spread' || name === 'then') {
+    return args.length === 2 ? 0 : -1
+  }
+
+  if (name === 'request') {
+    return args.length === 1 && _.isObject(args[0]) ? 0 : -1
+  }
+
+  if (name === 'wait') {
+    return args.length === 2 ? 1 : -1
+  }
+
+  if (name === 'nextUntil' || name === 'prevUntil' || name === 'parentsUntil') {
+    if (args.length === 2 && _.isObject(args[1])) {
+      return 1
+    }
```

Comment:
We have to keep up with this is there's ever a change in the options of these commands? Is there a test for this? If we change the options for these commands - we'll know to update this method? Seems brittle.

Reply:
* This part should be changed only when cypress API argument order changes or a new API is added whose last arg is an object but not an option. This function is created for those cases like `cy.spread(options, callbackFn)`. 

* The tests for this is located at `log_spec.js`'s `'last arg can be an object, but not an option'` section.